src/VisualStudio/Core/Def/Implementation/UnusedReferences/ProjectAssets/ProjectAssetsReaderService.cs

```diff
@@ -0,0 +1,52 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Immutable;
+using System.Composition;
+using System.IO;
+using Microsoft.CodeAnalysis.Host.Mef;
+using Microsoft.CodeAnalysis.UnusedReferences;
+using Microsoft.CodeAnalysis.UnusedReferences.ProjectAssets;
+using Newtonsoft.Json;
+
+namespace Microsoft.VisualStudio.LanguageServices.Implementation.UnusedReferences.ProjectAssets
+{
+    [ExportWorkspaceService(typeof(IProjectAssetsReaderService)), Shared]
+    internal sealed class ProjectAssetsReaderService : IProjectAssetsReaderService
+    {
+        [ImportingConstructor]
+        [Obsolete(MefConstruction.ImportingConstructorMessage, error: true)]
+        public ProjectAssetsReaderService()
+        {
+        }
+
+        /// <summary>
+        /// Enhances references with the assemblies they bring into the compilation and their dependency hierarchy.
+        /// </summary>
+        public ImmutableArray<ReferenceInfo> ReadReferences(
+            ImmutableArray<ReferenceInfo> projectReferences,
+            string projectAssetsFilePath)
+        {
+            if (!File.Exists(projectAssetsFilePath))
```

Comment:
1. this can throw.  can you use IOUtilities instead? 
2. this is doing IO, this function should be async.